server/src/main/java/io/druid/server/metrics/EventReceiverFirehoseMonitor.java

```diff
@@ -19,24 +19,34 @@
 
 package io.druid.server.metrics;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.inject.Inject;
 import com.metamx.emitter.service.ServiceEmitter;
 import com.metamx.emitter.service.ServiceMetricEvent;
 import com.metamx.metrics.AbstractMonitor;
+import com.metamx.metrics.KeyedDiff;
+import com.metamx.metrics.MonitorUtils;
+import io.druid.segment.realtime.firehose.EventReceiverFirehoseFactory;
 
 import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class EventReceiverFirehoseMonitor extends AbstractMonitor
 {
 
   private final EventReceiverFirehoseRegister register;
+  private final KeyedDiff keyedDiff = new KeyedDiff();
+  private final Map<String, String[]> dimensions;
 
   @Inject
   public EventReceiverFirehoseMonitor(
-      EventReceiverFirehoseRegister eventReceiverFirehoseRegister
+      EventReceiverFirehoseRegister eventReceiverFirehoseRegister,
+      Properties props
   )
   {
     this.register = eventReceiverFirehoseRegister;
+    this.dimensions = MonitorsConfig.getDimensions(props);
```

Comment:
this will recognize all props starting with "druid.metrics.emitter.dimension.", I understand that currently there are only two (dataSource and taskId) but in future we might add more and all of them will end up getting emitted from here unintentionally. Should we look for specific dimensions present instead of blindly taking all?